wn/lmf.py

```diff
@@ -90,12 +99,31 @@ def __iter__(self):
     def __next__(self):
         _next = self._next
         if _next == (None, None):
+            if self._progress_info:
+                self._progress_info.progress_handler.set(status="Completed")
             raise StopIteration
         self._next = next(self.iterator, (None, None))
         return _next
 
+    def _update_progress(self):
+        if self._progress_info:
+            self._progress += 1
+            # We don't want to refresh each cycle
+            refresh_cycle = 20
+            if self._progress % refresh_cycle == 0:
```

Comment:
I understand the performance reasons why we don't refresh at each iteration, but I think it'd be cleaner to move this logic to the `wn.util.ProgressHandler` and `ProgressBar` classes. E.g.,

```diff
class ProgressHandler:
    ...
    def __init__(
        self,
        *,
        message: str = '',
        count: int = 0,
        total: int = 0,
+       refresh_interval: int = 1,
        unit: str = '',
        status: str = '',
        file: TextIO = sys.stderr,
    ):
        self.file = file
        self.kwargs = {
            'count': count,
            'total': total,
+           'refresh_interval': refresh_interval,
            'message': message,
            'unit': unit,
            'status': status,
        }
+        self._refresh_quota: int = refresh_interval
...
class ProgressBar(ProgressHandler):
    ...
    def update(self, n: int = 1) -> None:
        """Increment the count by *n* and print the reformatted bar."""
        self.kwargs['count'] += n  # type: ignore
+       self._refresh_quota -= n
+       if self._refresh_quota <= 0:
+           self._refresh_quota = self.kwargs['refresh_interval']
+           s = self.format()
-       s = self.format()
-       ...
```

Reply:
Thanks, this one is pretty good actually.